src/NfcService.cpp

```diff
@@ -89,6 +89,8 @@ NfcManager* NfcService::sNfcManager = NULL;
 NfcService::NfcService()
  : mState(STATE_NFC_OFF)
  , mIsTagPresent(false)
+ , mTagSessionId(-1)
+ , mNotifyTagLost(false)
```

Comment:
It depends how we define the meaning of mIsTagPresent, currently it will be set to false immediately when PresenceCheck fail, if we want to merge these two variables then mIsTagPresent = false means PresenceCheck fail AND "tech lost is notified"
The drawback of this way is that if a tag is removed and then tag connected again before tech lost is notified, then the tech discover of next tag will not be triggered.
Please let me know how do you think about this.